tests/test_tls/test_tls.c

```diff
@@ -34,6 +34,11 @@ struct tcb {
     void *pad;
     volatile uint64_t _data;
 };
+#elif defined(__powerpc64__)
```

Comment:
This definition of `tcb` seems equivalent to that in `/* Variant II */` above, sans the extra `volatile`. Are you sure you need your own definition here and can't just re-use the common one for Variant II?

Reply:
Variant II is correct. I had to add the volatile in front of the `void *tp` because of this here:

```
CC test_tls.c
test_tls.c: In function ‘solo5_app_main’:
test_tls.c:76:13: error: assignment discards ‘volatile’ qualifier from pointer target type [-Werror=discarded-qualifiers]
     tcb1.tp = &tcb1._data;
             ^
test_tls.c:77:13: error: assignment discards ‘volatile’ qualifier from pointer target type [-Werror=discarded-qualifiers]
     tcb2.tp = &tcb2._data;
             ^
cc1: all warnings being treated as errors
```

Reply:
Just use a cast in the assignment then, better than duplicating the definition unnecessarily.